### PR TITLE
Generate better API documentation

### PR DESCRIPTION
**Pull request by ManasJayanth**
/cc @yurydelendik  @brendandahl 

Initiating a thread discussing ways to improving the docs and feedback for the same.
Starting with missing comments on the API docs page

/re #6526

### CODE REVIEW

File: src/display/api.js [hunk 1]
```diff
@@ -346,145 +346,143 @@ PDFJS.getDocument = function getDocument(src,
...
+       * Callback to be able to monitor the loading progress of the PDF file
+       * (necessary to implement e.g. a loading bar). The callback receives
+       * an {Object} with the properties: {number} loaded and {number} total.
+       */
+      this.onProgress = null;
     }
-  };
 
-  return PDFDocumentLoadingTask;
-})();
```
Comment (yurydelendik) [inline src/display/api.js]: Unneeded code change -- please revert tabulation back.

**Comment by yurydelendik** [comment]
Nice!

Can you also fix double entry of PDFDocumentLoadingTask (see http://107.22.172.223:8877/824a96518b07a1c/api/draft/PDFDocumentLoadingTask.html)? Interesting that PDFDocumentProxy (see http://107.22.172.223:8877/824a96518b07a1c/api/draft/PDFDocumentProxy.html) has no such issue.

**Comment by timvandermeij** [comment]
I agree, this is looking much better now! Fortunately the update to JSDoc 3.3.3 also included a better lay-out, so it is more pleasant to look at too.

**Comment by prometheansacrifice** [comment]
@timvandermeij Update JSDoc for PDFJS too?

**Comment by timvandermeij** [comment]
No, we already use JCDoc 3.3.3 as can be seen on the last text line of http://107.22.172.223:8877/824a96518b07a1c/api/draft/PDFDocumentLoadingTask.html. I just meant to say that the general look of the documentation has improved with that update from recently. No need to do any updates, sorry for the confusion :)

**Comment by prometheansacrifice** [comment]
@timvandermeij Oh! I thought you were referring to the version in [package.json](https://github.com/mozilla/pdf.js/blob/master/package.json#L5) which is 3.3.0-alpha9 Thank you :)

Anyways, about adding more examples within the docs, how do I get started?

File: src/display/api.js [hunk 2]
```diff
@@ -346,9 +346,9 @@ PDFJS.getDocument = function getDocument(src,
 /**
  * PDF document loading operation.
  * @class
+ * @alias PDFDocumentLoadingTask
  */
 var PDFDocumentLoadingTask = (function PDFDocumentLoadingTaskClosure() {
-  /** @constructs PDFDocumentLoadingTask */
```
Comment (yurydelendik) [inline src/display/api.js]: Can you remove other @constructs in this file too?

File: src/display/api.js [hunk 3]
```diff
@@ -405,13 +405,12 @@ var PDFDocumentLoadingTask = (function PDFDocumentLoadingTaskClosure() {
...
+ * @param {Uint8Array} initialData
  */
-var PDFDataRangeTransport = (function pdfDataRangeTransportClosure() {
-  /**
-   * @constructs PDFDataRangeTransport
-   * @param {number} length
-   * @param {Uint8Array} initialData
-   */
+var PDFDataRangeTransport = PDFJS.PDFDataRangeTransport =
+(function pdfDataRangeTransportClosure() {
```
Comment (yurydelendik) [inline src/display/api.js]: nit: wrong indenture,  break after `(function` with four spaces indent here 
(or just revert the change if possible)

Comment (prometheansacrifice) [inline src/display/api.js]: Like this?

```
var PDFDataRangeTransport = PDFJS.PDFDataRangeTransport =
    (function pdfDataRangeTransportClosure() {
```

Comment (yurydelendik) [inline src/display/api.js]: both of them are okay

File: src/display/api.js [hunk 4]
```diff
@@ -405,86 +405,85 @@ var PDFDocumentLoadingTask = (function PDFDocumentLoadingTaskClosure() {
...
-        }
-      }.bind(this));
-    },
-
-    onDataProgressiveRead:
-        function PDFDataRangeTransport_onDataProgress(chunk) {
-      this._readyCapability.promise.then(function () {
-        var listeners = this._progressiveReadListeners;
+var PDFDataRangeTransport = PDFJS.PDFDataRangeTransport =
+  (function pdfDataRangeTransportClosure() {
```
Comment (Snuffleupagus) [inline src/display/api.js]: I believe that comment https://github.com/mozilla/pdf.js/pull/6599#discussion-diff-44352636 meant that *only* this line should be indented with *four*&nbsp;spaces, and that the lines below should not be touched.

Comment (prometheansacrifice) [inline src/display/api.js]: @yurydelendik Is that so?

Comment (yurydelendik) [inline src/display/api.js]: right, you shall touch only affected code:

```
var PDFDataRangeTransport = PDFJS.PDFDataRangeTransport =
    (function pdfDataRangeTransportClosure() {
```

rest of the code shall stay intact.

See also examples at https://github.com/mozilla/pdf.js/wiki/Style-Guide#classes

File: src/display/api.js [hunk 5]
```diff
@@ -422,19 +421,19 @@ var PDFDataRangeTransport = (function pdfDataRangeTransportClosure() {
     this._readyCapability = createPromiseCapability();
   }
   PDFDataRangeTransport.prototype =
-      /** @lends PDFDataRangeTransport.prototype */ {
+  /** @lends PDFDataRangeTransport.prototype */ {
```
Comment (Snuffleupagus) [inline src/display/api.js]: Please revert this, and all the other below, indentation changes.

**Comment by prometheansacrifice** [comment]
Back to how it was before along with the fix

**Comment by yurydelendik** [comment]
Looks good. Thank you for the patch.
